src/Camelot.Services.AllPlatforms/ShellIconsCacheService.cs

```diff
@@ -0,0 +1,180 @@
+using System;
+using System.Collections.Generic;
+using Camelot.Services.Abstractions;
+using System.IO;
+using Camelot.Services.Abstractions.Models;
+using Camelot.Images;
+using Avalonia.Media.Imaging;
+using Camelot.Services.Environment.Interfaces;
+using Camelot.Services.Environment.Enums;
+
+namespace Camelot.Services.AllPlatforms;
+
+public class ShellIconsCacheService : IShellIconsCacheService
```

Comment:
I think we should leave "Cache" part of the name for this class out because we don't really want to expose to consumers such details as using cache internally. It seems that we use `ShellIconsService` only here so moving all code to the same file sounds fine to me

Reply:
ShellIconsService is per platform, just asking platform for icon, based on file extension or application.
ShellIconsCacheService is higher level, shared to all platform, that checks if it already has icon for extension.
eg for file1.txt and file2.txt,  ShellIconsCacheService  will call ShellIconsService  only on 1st time.
on 2nd time, it will use cache.

Reply:
I got the idea but I don't like the naming, we should not let consumers of this class know that there is some cache being used. I recommend to implement this caching logic as a decorator or put it in some base class and for platforms just add some platform-specific implementation